synthtool/gcp/common.py

```diff
@@ -153,10 +153,22 @@ def _load_partials(self, metadata: Dict):
 
 def decamelize(str: str):
     """ parser to convert fooBar.js to Foo Bar. """
-    str2 = str[0].upper()
-    for chr in str[1:]:
-        if re.match(r"[A-Z]", chr):
-            str2 += " " + chr.upper()
+    prev = " "
+    str2 = ""
+    for i in range(0, len(str)):
+        chr = str[i]
+        if prev == " ":
+            str2 += chr.upper()
+        elif re.match(r"[A-Z]", chr):
+            # handle "JSONFrom" -> "JSON From" case.
+            next = ""
```

Comment:
`next` is also a built-in, so it's best not to overwrite it. Usually we use a trailing underscore to avoid these kinds of collisions. `next_`.